File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/loader/TokenLoader.java
package ru.usedesk.chat_sdk.internal.data.framework.loader;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import javax.inject.Inject;
import javax.inject.Named;
import javax.inject.Singleton;

import ru.usedesk.chat_sdk.internal.data.framework.info.DataLoader;

@Singleton
public class TokenLoader extends DataLoader<String> {
    private static final String PREF_NAME = "usedeskSdkToken";
    private static final String KEY_TOKEN = "token";

    private final SharedPreferences sharedPreferences;

    @Inject
    @Named("token")
    TokenLoader(@NonNull Context context) {
        this.sharedPreferences = context.getSharedPreferences(PREF_NAME, Context.MODE_MULTI_PROCESS);
    }

    @Nullable
    @Override
    protected String loadData() {
        return sharedPreferences.getString(KEY_TOKEN, null);
    }

    @Override
    protected void saveData(@NonNull String token) {
        sharedPreferences.edit()
                .putString(KEY_TOKEN, token)
                .apply();
    }

    @Override
    public void clearData() {
        sharedPreferences.edit()
                .remove(KEY_TOKEN)
                .apply();
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/token/TokenLoader.kt
package ru.usedesk.chat_sdk.internal.data.framework.token

import android.content.Context
import android.content.SharedPreferences
import ru.usedesk.chat_sdk.internal.data.framework.info.DataLoader
import toothpick.InjectConstructor

@InjectConstructor
class TokenLoader(
        context: Context
) : DataLoader<String>(), ITokenLoader {

    private val sharedPreferences: SharedPreferences = context.getSharedPreferences(
            PREF_NAME,
            Context.MODE_PRIVATE
    )

    override fun loadData(): String? {
        return sharedPreferences.getString(KEY_TOKEN, null)
    }

    override fun saveData(data: String) {
        sharedPreferences.edit()
                .putString(KEY_TOKEN, data)
                .apply()
    }

    override fun clearData() {
        super.clearData()
        sharedPreferences.edit()
                .remove(KEY_TOKEN)
                .apply()
    }

    companion object {
        private const val PREF_NAME = "usedeskSdkToken"
        private const val KEY_TOKEN = "token"
    }
}